From 5faec99b43ecbc896ebfef8905ad3510f8ea4af8 Mon Sep 17 00:00:00 2001
Message-Id: <5faec99b43ecbc896ebfef8905ad3510f8ea4af8.1424982292.git.jen@redhat.com>
In-Reply-To: <f81cd7732c9ee920b054762eae107cff8126f3df.1424982292.git.jen@redhat.com>
References: <f81cd7732c9ee920b054762eae107cff8126f3df.1424982292.git.jen@redhat.com>
From: Alex Williamson <alex.williamson@redhat.com>
Date: Wed, 25 Feb 2015 19:02:00 -0500
Subject: [CHANGE 3/3] Use spec compliant timeouts
To: rhvirt-patches@redhat.com,
    jen@redhat.com

RH-Author: Alex Williamson <alex.williamson@redhat.com>
Message-id: <20150225190200.17227.74500.stgit@gimli.home>
Patchwork-id: 63940
O-Subject: [RHEL6.7 gpxe PATCH 3/3] [RHEL-only][config/local/dhcp] Use spec compliant timeouts
Bugzilla: 968474
RH-Acked-by: Jeff Nelson <jenelson@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>

Bugzilla: 968474
Upstream: N/A

Use local config to override gPXE's abbreviated DHCP timeouts using
the recommended values for spec compliance.

Signed-off-by: Alex Williamson <alex.williamson@redhat.com>
---
 src/config/local/dhcp.h |   52 +++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 52 insertions(+)
 create mode 100644 src/config/local/dhcp.h

Signed-off-by: Jeff E. Nelson <jen@redhat.com>
---
 src/config/local/dhcp.h | 52 +++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 52 insertions(+)
 create mode 100644 src/config/local/dhcp.h

diff --git a/src/config/local/dhcp.h b/src/config/local/dhcp.h
new file mode 100644
index 0000000..027873d
--- /dev/null
+++ b/src/config/local/dhcp.h
@@ -0,0 +1,52 @@
+/*
+ * Downstream localization
+ *
+ * For RHEL, use spec compliant DHCP timeouts (bz968474)
+ */
+
+/*
+ * PXE spec defines timeouts of 4, 8, 16, 32 seconds
+ */
+#undef DHCP_DISC_START_TIMEOUT_SEC
+#define DHCP_DISC_START_TIMEOUT_SEC 4
+#undef DHCP_DISC_END_TIMEOUT_SEC
+#define DHCP_DISC_END_TIMEOUT_SEC   32
+
+/*
+ * Elapsed time used for early break waiting for ProxyDHCP, this therefore
+ * needs to be less than the cumulative time for the first 2 timeouts.
+ */
+#undef DHCP_DISC_PROXY_TIMEOUT_SEC
+#define DHCP_DISC_PROXY_TIMEOUT_SEC 11
+
+/*
+ * Approximate PXE spec requirement using minimum timeout (0.25s) for
+ * timeouts of 0.25, 0.5, 1, 2, 4
+ */
+#undef DHCP_REQ_START_TIMEOUT_SEC
+#define DHCP_REQ_START_TIMEOUT_SEC  0
+#undef DHCP_REQ_END_TIMEOUT_SEC
+#define DHCP_REQ_END_TIMEOUT_SEC    4
+
+/*
+ * Same as normal request phase, except non-fatal, so we extend the timer
+ * to 8 and set the early timeout to an elapsed time value that causes a
+ * break after the 4 second timeout.
+ */
+#undef DHCP_PROXY_START_TIMEOUT_SEC
+#define DHCP_PROXY_START_TIMEOUT_SEC        0
+#undef DHCP_PROXY_END_TIMEOUT_SEC
+#define DHCP_PROXY_END_TIMEOUT_SEC  8
+#undef DHCP_REQ_PROXY_TIMEOUT_SEC
+#define DHCP_REQ_PROXY_TIMEOUT_SEC  7
+
+/*
+ * Same as above, retry each server using standard timeouts, extended by
+ * one so that we can increment to the next before a timer induced failure.
+ */
+#undef PXEBS_START_TIMEOUT_SEC
+#define PXEBS_START_TIMEOUT_SEC             0
+#undef PXEBS_END_TIMEOUT_SEC
+#define PXEBS_END_TIMEOUT_SEC               8
+#undef PXEBS_MAX_TIMEOUT_SEC
+#define PXEBS_MAX_TIMEOUT_SEC               7
-- 
2.1.0

